src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/commands.py

```diff
@@ -44,21 +47,26 @@ def _dont_fail_not_exist(ex):
         raise ex
 
 
+transform_keys_result = lambda x: x.keys if hasattr(x, 'keys') else x  # noqa: E731 lambda vs def
+
 # storage account commands
 factory = lambda kwargs: storage_client_factory().storage_accounts  # noqa: E731 lambda vs def
 cli_command(__name__, 'storage account check-name', mgmt_path + 'check_name_availability', factory)
 cli_command(__name__, 'storage account delete', mgmt_path + 'delete', factory, confirmation=True)
 cli_command(__name__, 'storage account show', mgmt_path + 'get_properties', factory, exception_handler=empty_on_404)
-cli_command(__name__, 'storage account create', custom_path + 'create_storage_account')
 cli_command(__name__, 'storage account list', custom_path + 'list_storage_accounts')
 cli_command(__name__, 'storage account show-usage', custom_path + 'show_storage_account_usage')
 cli_command(__name__, 'storage account show-connection-string', custom_path + 'show_storage_account_connection_string')
-cli_command(__name__, 'storage account keys renew', mgmt_path + 'regenerate_key', factory, transform=lambda x: x.keys)
-cli_command(__name__, 'storage account keys list', mgmt_path + 'list_keys', factory, transform=lambda x: x.keys)
-cli_generic_update_command(__name__, 'storage account update',
-                           mgmt_path + 'get_properties',
-                           mgmt_path + 'update', factory,
-                           custom_function_op=custom_path + 'update_storage_account')
+cli_command(__name__, 'storage account keys renew', mgmt_path + 'regenerate_key', factory, transform=transform_keys_result)
+cli_command(__name__, 'storage account keys list', mgmt_path + 'list_keys', factory, transform=transform_keys_result)
+
+cli_command(__name__, 'storage account create', custom_nonce_path + 'create_storage_account')
+if get_api_version(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS) in ['2016-12-01']:
+    cli_generic_update_command(__name__, 'storage account update',
+                               mgmt_path + 'get_properties',
+                               mgmt_path + 'update', factory,
+                               custom_function_op=custom_nonce_path + 'update_storage_account')
```

Comment:
Does this API call not exist in previous versions or is this an example only? This illustrates why I would prefer "api_version" over "get_api_version", but I would actually prefer something that looked like:
```Python
cli_generic_update_command(__name__, 'storage account update', 
   mgmt_path + 'get_properties', 
   mgmt_path + 'update', factory, 
   custom_function_op=custom_nonce_path + 'update_storage_account',
   min_api_version='foo', max_api_version='bar') 
```
Where min and max have defaults. The commands.py files are (in most modules) very straightforward and easy to read, and this preserves that. Adding conditionals makes them more difficult to maintain. Also, by putting this as part of the command, you can see the supported API versions in the command_table metadata. With if statements, you will not have that information.